infra/Pos/Subscription/Common.hs

```diff
@@ -60,17 +65,34 @@ data SubscriptionTerminationReason =
 -- giving the reason. Notices will be logged before and after the subscription.
 subscribeTo
     :: forall m. (SubscriptionMode m)
-    => SendActions m -> NodeId -> m SubscriptionTerminationReason
-subscribeTo sendActions peer = do
+    => Timer -> SendActions m -> NodeId -> m SubscriptionTerminationReason
+subscribeTo keepAliveTimer sendActions peer = do
     logNotice $ msgSubscribingTo peer
-    outcome <- try $ withConnectionTo sendActions peer $ \_peerData ->
-           pure $ Conversation $ \conv -> do
-               send conv MsgSubscribe
-               recv conv 0 :: m (Maybe Void) -- Other side will never send
+    outcome <- try $ withConnectionTo sendActions peer $ \_peerData -> NE.fromList
+        -- Sort conversations in descending order based on their version so that
+        -- the highest available version of the conversation is picked.
+        [ Conversation convMsgSubscribe
+        , Conversation convMsgSubscribe1
+        ]
     let reason = either Exceptional (maybe Normal absurd) outcome
     logNotice $ msgSubscriptionTerminated peer reason
     return reason
   where
+    convMsgSubscribe :: ConversationActions MsgSubscribe Void m -> m t
+    convMsgSubscribe conv = do
+        send conv MsgSubscribe
+        forever $ do
+            startTimer keepAliveTimer
```

Comment:
Is it OK if we have the same timer in multiple subscriptions?

Reply:
I was wondering about that.

It's reasonable in a sense that if we're not receiving blocks, then none of the subscriptions work, so we can recover. But if any of the subscriptions is active, then we won't detect inactive ones.